haigha/connection.py

```diff
@@ -404,7 +404,10 @@ def read_frames(self):
         # Send a heartbeat (if needed)
         self._channels[0].send_heartbeat()
 
-        data = self._transport.read(self._heartbeat)
+        # Wait for 2 heartbeat intervals before giving up. See AMQP 4.2.7:
+        # "If a peer detects no incoming traffic (i.e. received octets) for two heartbeat intervals or longer,
+        # it should close the connection"
+        data = self._transport.read(self._heartbeat*2)
```

Comment:
If I recall correctly, the value of `Connection._heartbeat` reflects the negotiation performed via connection tuning handshake. If the tuning handshake decides against heartbeats, then they should not be sent.